### PR TITLE
HPCC-17751 Definition Explorer/Details Widget

### PR DESCRIPTION
**Pull request by miguelvazq**
We can now show the XML tied to each definition in this interface. This interface is better suited for embedding into the IDE for DESDL. The bookmark should follow the following syntax.

<protocol://host:port>/?Widget=DynamicESDLDefinitionQueryWidget

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browers
  - [x] The component(s) render as expected

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith Please review. 
@rpastrana Please review translations
@g-pan Will need documentation

**Comment by rpastrana** [comment]
@miguelvazq ES translations look fine.

**Comment by miguelvazq** [comment]
@GordonSmith bump.

File: esp/src/eclwatch/DynamicESDLDefinitionQueryWidget.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
+/*##############################################################################
+#    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems®.
```
Comment (GordonSmith) [inline esp/src/eclwatch/DynamicESDLDefinitionQueryWidget.js]: Date wrong and I don't think we need to copyright header once the licence file is present in the repo.

File: esp/src/eclwatch/DynamicESDLDefinitionQueryWidget.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            var context = this;
+            var retVal = new declare([ESPUtil.Grid(false, true)])({
+                store: this.store,
+                columns: {
+                    Name: {label: this.i18n.Name, sortable: true, width:200}
+                }
+            }, domID);
+
+            retVal.on("dgrid-select", function (evt) {
+                var selection = context.grid.getSelected();
```
Comment (GordonSmith) [inline esp/src/eclwatch/DynamicESDLDefinitionQueryWidget.js]: Should really test if there was a selection (it might have been a de-selection)

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Comment by miguelvazq** [comment]
@GordonSmith Modified.

**Comment by GordonSmith** [comment]
Looks fine.

**Review by GordonSmith — APPROVED** [review]

**Comment by rpastrana** [comment]
@richardkchapman just as a follow up since this is late in the RC time-frame, this change is isolated, doesn't cause any work-flow interruptions, provides a significant usability improvement, and supports recent ECLIDE 6.4 changes

File: esp/src/eclwatch/nls/es/hpcc.js [hunk 3]
```diff
@@ -652,6 +652,8 @@ define(
     title_DFUQuery: "Archivos Lógicos",
     title_DFUWUDetails: "Unidad de trabajo DFU",
     title_DirectoriesFor: "Directorios para",
+    title_Definitions: "Definiciónes",
```
Comment (richardkchapman) [inline esp/src/eclwatch/nls/es/hpcc.js]: We will need to cycle around the other translations again...

Comment (GordonSmith) [inline esp/src/eclwatch/nls/es/hpcc.js]: @miguelvazq FYI the translations.

**Pull request merged**